biojava-structure/src/main/java/org/biojava/nbio/structure/geometry/SuperPositionQCP.java

```diff
@@ -0,0 +1,441 @@
+ * terms of the GNU Lesser General Public Licence.  This should
+ * be distributed with the code.  If you do not have a copy,
+ * see:
+ *
+ *      http://www.gnu.org/copyleft/lesser.html
+ *
+ * Copyright for this code is held jointly by the individual
+ * authors.  These should be listed in @author doc comments.
+ *
+ * For more information on the BioJava project and its aims,
+ * or to join the biojava-l mailing list, visit the home page
+ * at:
+ *
+ *      http://www.biojava.org/
+ *
+ */
+
+package org.biojava.nbio.structure.geometry;
+
+import javax.vecmath.Matrix3d;
+import javax.vecmath.Matrix4d;
+import javax.vecmath.Point3d;
+import javax.vecmath.Vector3d;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of the Quaternion-Based Characteristic Polynomial algorithm
+ * for RMSD and Superposition calculations.
+ * <p>
+ * Reference: Theobald DL (2005). Rapid calculation of RMSDs using a
+ * quaternion-based characteristic polynomial. Acta Crystallogr A. 2005
+ * Jul;61(Pt 4):478-80. (PubMed id: 15973002)
+ * 
+ * @author Peter Rose
+ * @author Aleix Lafita
+ * 
+ */
+public final class SuperPositionQCP {
+	
+	private static final Logger logger = LoggerFactory
+			.getLogger(SuperPositionQCP.class);
+
+	double evecprec = 1E-6;
+	double evalprec = 1E-11;
+
+	private Point3d[] x = null;
+	private Point3d[] y = null;
+
+	private double[] weight = null;
+
+	private Point3d[] xref = null;
+	private Point3d[] yref = null;
+	private Point3d xtrans;
+	private Point3d ytrans;
+
+	private double e0;
+	private Matrix3d rotmat = new Matrix3d();
+	private Matrix4d transformation = new Matrix4d();
+	private double rmsd = 0;
+	private double Sxy, Sxz, Syx, Syz, Szx, Szy;
+	private double SxxpSyy, Szz, mxEigenV, SyzmSzy, SxzmSzx, SxymSyx;
+	private double SxxmSyy, SxypSyx, SxzpSzx;
+	private double Syy, Sxx, SyzpSzy;
+	private boolean rmsdCalculated = false;
+	private boolean transformationCalculated = false;
+	private boolean centered = false;
+
+	public void set(Point3d[] x, Point3d[] y) {
+		this.x = x;
+		this.y = y;
+		rmsdCalculated = false;
+		transformationCalculated = false;
+	}
+
+	public void set(Point3d[] x, Point3d[] y, double[] weight) {
+		this.x = x;
+		this.y = y;
+		this.weight = weight;
+		rmsdCalculated = false;
+		transformationCalculated = false;
+	}
+
+	public void setCentered(boolean centered) {
+		this.centered = centered;
+	}
+
+	public double getRmsd() {
+		if (!rmsdCalculated) {
+			calcRmsd(x, y);
+		}
+		return rmsd;
+	}
+
+	public Matrix4d getTransformationMatrix() {
+		getRotationMatrix();
+		if (!centered) {
+			calcTransformation();
+		} else {
+			transformation.set(rotmat);
+		}
+		return transformation;
+	}
+
+	public Matrix3d getRotationMatrix() {
+		getRmsd();
+		if (!transformationCalculated) {
+			calcRotationMatrix();
+		}
+		return rotmat;
+	}
+
+	public Point3d[] getTransformedCoordinates() {
+		CalcPoint.transform(transformation, x);
```

Comment:
Comparing this to https://github.com/rcsb/QuickStructureSearch/blob/master/src/main/java/org/rcsb/structuralAlignment/SuperPositionQCP.java there's 2 important differences:
- the coordinates that are moved are `y` instead of `x`
- the moved coordinates are cloned into a new array instead of transforming the input

It seems like in the [original C code](http://theobald.brandeis.edu/qcp/qcprot.c) `y` are the coordinates that move.

If this is not covered by a test we should try to write a test for this to make sure it's working correctly comparing to other superposition methods.